tests/unit/test_table_options.py

```diff
@@ -104,111 +104,151 @@ def test_table_clustering_fields_dialect_option_type_error(faux_conn):
+    YEAR. BigQuery cannot partition on DATE by HOUR, so that is expected to
+    xfail.
 
-    assert " ".join(faux_conn.test_data["execute"][-1][0].strip().split()) == (
-        "CREATE TABLE `some_table` ( `id` INT64, `createdAt` DATETIME )"
-        " PARTITION BY DATE_TRUNC(createdAt, DAY)"
-    )
+    IF time_partitioning_type is None, the __init__() in TimePartitioning will
+    overwrite it with TimePartitioningType.DAY as the default.
+    """
 
+    if time_partitioning_type is None:
+        time_partitioning_type = TimePartitioningType.DAY
 
-def test_table_time_partitioning_by_month_dialect_option(faux_conn):
-    # expect table creation to fail as SQLite does not support partitioned tables
     with pytest.raises(sqlite3.OperationalError):
         setup_table(
             faux_conn,
             "some_table",
             sqlalchemy.Column("id", sqlalchemy.Integer),
-            sqlalchemy.Column("createdAt", sqlalchemy.DateTime),
+            sqlalchemy.Column("createdAt", column_dtype),
             bigquery_time_partitioning=TimePartitioning(
-                field="createdAt",
-                type_=TimePartitioningType.MONTH,
+                field="createdAt", type_=time_partitioning_type
             ),
         )
 
-    assert " ".join(faux_conn.test_data["execute"][-1][0].strip().split()) == (
-        "CREATE TABLE `some_table` ( `id` INT64, `createdAt` DATETIME )"
-        " PARTITION BY DATE_TRUNC(createdAt, MONTH)"
+    result = " ".join(faux_conn.test_data["execute"][-1][0].strip().split())
+    expected = (
+        f"CREATE TABLE `some_table` ( `id` INT64, `createdAt` {column_dtype.__visit_name__} )"
+        f" PARTITION BY {func_name}(createdAt, {time_partitioning_type})"
     )
+    assert result == expected
 
 
-def test_table_time_partitioning_with_timestamp_dialect_option(faux_conn):
-    # expect table creation to fail as SQLite does not support partitioned tables
+def test_table_time_partitioning_with_field_dialect_option(faux_conn):
+    """Expect table creation to fail as SQLite does not support partitioned tables
+
+    Confirms that if the column datatype is DATETIME but no TimePartitioning.type_
+    has been supplied, the system will default to DAY.
+    """
+
     with pytest.raises(sqlite3.OperationalError):
         setup_table(
             faux_conn,
             "some_table",
             sqlalchemy.Column("id", sqlalchemy.Integer),
-            sqlalchemy.Column("createdAt", sqlalchemy.TIMESTAMP),
+            sqlalchemy.Column("createdAt", sqlalchemy.DateTime),
             bigquery_time_partitioning=TimePartitioning(field="createdAt"),
         )
-
-    assert " ".join(faux_conn.test_data["execute"][-1][0].strip().split()) == (
-        "CREATE TABLE `some_table` ( `id` INT64, `createdAt` TIMESTAMP )"
-        " PARTITION BY TIMESTAMP_TRUNC(createdAt, DAY)"
+    result = " ".join(faux_conn.test_data["execute"][-1][0].strip().split())
```

Comment:
It seems the second half of this test does the following, as the docstring suggests:
```
Confirms that if the column datatype is DATETIME but no TimePartitioning.type_
    has been supplied, the system will default to DAY.
```
Maybe I'm missing something, but isn't this covered in the test case in line 136?

Reply:
The difference between this test and the parametrized test above is minor.

Essentially, in the parameters on line 136 we explicitly provide a value of `None` as a `type_` argument to the TimePartitioning class.

We know the default is `None` in `python-bigquery` but we set the value deliberately.

In line 193, we do not set the value of `type_` explicitly, we allow the default value (which should be `None`) come through and thus confirm that everything works as expected whether we provide a value to `type_` or not.